### PR TITLE
Hotfix: moving pinned items 

### PR DESCRIPTION
**Pull request by negue**
This fixes / increases / refactors the checks on moving items

2nd try #12932

### CODE REVIEW

**Comment by paglias** [comment]
LGTM!

File: website/server/controllers/api-v3/user.js [hunk 1]
```diff
@@ -1727,17 +1727,25 @@ api.movePinnedItem = {
       user.pinnedItemsOrder = currentPinnedItems.map(item => item.path);
     }
 
+    const itemExistInPinnedArray = user.pinnedItems.findIndex(item => item.path === path);
+
+    if (itemExistInPinnedArray === -1) {
+      throw new BadRequest(res.t('wrongItemPath', { path }, req.language));
```
Comment (Alys) [inline website/server/controllers/api-v3/user.js:1736]: Will this cause a bug when a player tries to move one of the automatically pinned items that isn't in their `pinnedItems` array?

Comment (paglias) [reply website/server/controllers/api-v3/user.js:1736]: Oh I thought this had been merged but given this doubt let's wait for @negue

Comment (negue) [reply website/server/controllers/api-v3/user.js:1736]: good point! gonna rework this a bit and try to add test for that as well - cause yes at that point that item isn't in user list (atleast not in the database array) the result however has those items .... so lets see how do it properly ^^

Comment (negue) [reply website/server/controllers/api-v3/user.js:1736]: the new push should allow to move official items too

Comment (negue) [reply website/server/controllers/api-v3/user.js:1736]: or not... `import getOfficialPinnedItems from '../../../common/script/libs/getOfficialPinnedItems';` can't be used in the user controller ..? any workaround to get this there too ?

Edit: using only the `common` import it worked out :)

**Comment by SabreCat** [comment]
Looks good on my run through!

**Pull request merged**